test/relayserver/RegistrationManager.test.ts

```diff
@@ -323,40 +342,67 @@ contract('RegistrationManager', function (accounts) {
         assert.isTrue(newServer.registrationManager.isHubAuthorized, 'Hub should be authorized in server')
+        await evmMineMany(unstakeDelay)
+
         const latestBlock = await env.web3.eth.getBlock('latest')
         const receipts = await newServer._worker(latestBlock.number)
+
         assert.isFalse(newServer.registrationManager.isHubAuthorized, 'Hub should not be authorized in server')
         const gasPrice = await env.web3.eth.getGasPrice()
         // TODO: these two hard-coded indexes are dependent on the order of operations in 'withdrawAllFunds'
-        const workerEthTxCost = getTotalTxCosts([receipts[1]], gasPrice)
-        const managerHubSendTxCost = getTotalTxCosts([receipts[0]], gasPrice)
+        const workerEthTxCost = getTotalTxCosts([receipts[0]], gasPrice)
         const ownerBalanceAfter = toBN(await env.web3.eth.getBalance(relayOwner))
         const managerHubBalanceAfter = await env.relayHub.balanceOf(newServer.managerAddress)
         const managerBalanceAfter = await newServer.getManagerBalance()
         const workerBalanceAfter = await newServer.getWorkerBalance(workerIndex)
-        assert.isTrue(managerHubBalanceAfter.eqn(0))
+        assert.equal(managerHubBalanceAfter.toString(), managerHubBalanceBefore.toString())
         assert.isTrue(workerBalanceAfter.eqn(0))
-        assert.equal(managerBalanceAfter.toString(), managerBalanceBefore.sub(managerHubSendTxCost).toString())
+        assert.equal(managerBalanceAfter.toString(), managerBalanceBefore.toString())
         assert.equal(
           ownerBalanceAfter.sub(
             ownerBalanceBefore).toString(),
-          managerHubBalanceBefore.add(workerBalanceBefore).sub(workerEthTxCost).toString(),
+          workerBalanceBefore.sub(workerEthTxCost).toString(),
           `ownerBalanceAfter(${ownerBalanceAfter.toString()}) - ownerBalanceBefore(${ownerBalanceBefore.toString()}) != 
-         managerHubBalanceBefore(${managerHubBalanceBefore.toString()}) + workerBalanceBefore(${workerBalanceBefore.toString()})
+         workerBalanceBefore(${workerBalanceBefore.toString()})
          - workerEthTxCost(${workerEthTxCost.toString()})`)
       })
     })
 
     it('_handleStakedEvent')
   })
 
+  describe('#_extractDuePendingEvents', () => {
+    let rm: RegistrationManager
+    let extracted: any[]
+
+    before(async () => {
+      // @ts-ignore
+      if (!relayServer.initialized) await relayServer.init()
+
+      rm = relayServer.registrationManager;
+
+      (rm as any).delayedEvents = [
+        { block: 1, eventData: 'event1' },
+        { block: 2, eventData: 'event2' },
+        { block: 3, eventData: 'event3' }
+      ]
+      extracted = rm._extractDuePendingEvents(2) as any
+    })
+    it('should extract events which are due (lower or equal block number)', function () {
+      assert.deepEqual(extracted, ['event1', 'event2'])
+    })
+    it('should should future events in the delayedEvents list', function () {
```

Comment:
should should

Reply:
I should should proof proof read read my my texts texts...